ambry-router/src/main/java/com/github/ambry/router/GetBlobInfoOperation.java

```diff
@@ -326,28 +329,44 @@ private void processGetBlobInfoResponse(GetRequestInfo getRequestInfo, GetRespon
             // any server error code that is not equal to ServerErrorCode.No_Error, the onErrorResponse should be invoked
             // because the operation itself doesn't succeed although the response in some cases is successful (i.e. Blob_Deleted)
             onErrorResponse(getRequestInfo.replicaId,
-                new RouterException("Server returned: " + getError, routerErrorCode));
+                new RouterException("Server returned: " + getError, routerErrorCode), true);
           }
         }
       }
     } else {
       logger.trace("Replica {} returned an error {} for a GetBlobInfoRequest with response correlationId : {} ",
           getRequestInfo.replicaId.getDataNodeId(), getError, getResponse.getCorrelationId());
-      onErrorResponse(getRequestInfo.replicaId, new RouterException("Server returned", processServerError(getError)));
+      onErrorResponse(getRequestInfo.replicaId, new RouterException("Server returned", processServerError(getError)),
+          true);
     }
   }
 
+  /**
+   * Process response if it was rejected due to quota compliance.
+   * @param correlationId correlation id of the request.
+   * @param replicaId {@link ReplicaId} of the request.
+   */
+  private void processQuotaRejectedResponse(int correlationId, ReplicaId replicaId) {
+    logger.trace("GetBlobInfoRequest with response correlationId {} was rejected because quota was exceeded.",
+        correlationId);
+    onErrorResponse(replicaId, new RouterException("QuotaExceeded", RouterErrorCode.TooManyRequests), false);
+    checkAndMaybeComplete();
```

Comment:
let's keep it consistent cross all operations.
We can do a couple things in the `processQuotaRejectedResponse`
1. call onErrorResponse
2. set operationComplete to true. in the GetBlobOperation, we should call setOperationCompleted.
3. call checkAndMaybeComplete();

Reply:
done.